### PR TITLE
more work on stops

### PR DESCRIPTION
**Pull request by hjanetzek**
I've changed our tilesize dependent width to use pixel as units. This implements one of the supported width units of tangram-js and was also somewhat necessary to visually validate that the width interpolation with stops works correctly :)

### CODE REVIEW

**Comment by tallytalwar** [comment]
Reviewing this now.

File: core/src/scene/stops.cpp [hunk 1]
```diff
@@ -0,0 +1,84 @@
+#include "stops.h"
+#include "csscolorparser.hpp"
+#include "yaml-cpp/yaml.h"
+#include <cassert>
+#include <algorithm>
+
+#define RED(col) (col % 256)
```
Comment (tallytalwar) [inline core/src/scene/stops.cpp]: Thoughts?

We should move these define to the header?

Comment (blair1618) [inline core/src/scene/stops.cpp]: A utility file with higher-level functions on colors would be useful in many places, distance conversions too; just not macros in headers :P

Comment (tallytalwar) [inline core/src/scene/stops.cpp]: Yes, I got rid of that :D. (hence ... ~~..~~)

**Pull request merged**